lightning/src/sync/debug_sync.rs

```diff
@@ -77,7 +77,7 @@ fn get_construction_location(backtrace: &Backtrace) -> String {
 	// Find the first frame that is after `debug_sync` (or that is in our tests) and use
 	// that as the mutex construction site. Note that the first few frames may be in
 	// the `backtrace` crate, so we have to ignore those.
-	let sync_mutex_constr_regex = regex::Regex::new(r"lightning.*debug_sync.*new").unwrap();
+	let sync_mutex_constr_regex = regex::Regex::new(r"lightning.*debug_sync").unwrap();
```

Comment:
Do we need a trailing `.*` to match the locks type (and `new` when it is not truncated)?